handlers/whatsapp/whatsapp.go

```diff
@@ -106,11 +147,42 @@ func (h *handler) receiveEvent(ctx context.Context, channel courier.Channel, w h
 			return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
 		}
 
-		// TODO: deal with media messages
-		// TODO: deal with location messages
+		text := ""
+		mediaURL := ""
+
+		if msg.Type == "text" {
+			text = msg.Text.Body
+		} else if msg.Type == "audio" {
+			mediaURL, err = resolveMediaURL(channel, msg.Audio.ID)
+		} else if msg.Type == "document" {
+			text = msg.Document.Caption
+			mediaURL, err = resolveMediaURL(channel, msg.Document.ID)
+		} else if msg.Type == "image" {
+			text = msg.Image.Caption
+			mediaURL, err = resolveMediaURL(channel, msg.Image.ID)
+		} else if msg.Type == "location" {
+			mediaURL = fmt.Sprintf("geo:%f,%f", msg.Location.Latitude, msg.Location.Longitude)
+		} else if msg.Type == "video" {
+			mediaURL, err = resolveMediaURL(channel, msg.Video.ID)
+		} else if msg.Type == "voice" {
+			mediaURL, err = resolveMediaURL(channel, msg.Voice.ID)
+		} else {
+			// we received a message type we do not support.
+			handlers.WriteAndLogRequestError(ctx, h, channel, w, r, fmt.Errorf("Unsupported message type %s", msg.Type))
```

Comment:
I think to log only you need `courier.LogRequestError` or `courier.LogRequestIgnored` the response in this case is not needed